### PR TITLE
Trusty service: Build old runner jar

### PR DESCRIPTION
**Pull request by radtriste**
We need the runner jar for images.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Review by manstis — COMMENTED** [review]
Hi @radtriste do you know why this change is needed? I [purposefully](https://github.com/kiegroup/kogito-apps/pull/1198/files) set the packaging type to `fast-jar` to allow me to use SNAPSHOTs locally instead of images deployed through `kogito-images` to quay.io. Without it, the JAR built by Quarkus, could not be used as a local container image... I'd have to dig again to remind myself of the exact technical details.

**Comment by radtriste** [comment]
For now we need the runner jars to be deployed to Nexus to build the images, else it will just fail ...
There is a refactoring we should be done in the near future to build images directly from the kogito-apps repo but it is not done yet. So this PR is needed for nightly/release of images

**Comment by radtriste** [comment]
> Without it, the JAR built by Quarkus, could not be used as a local container image... I'd have to dig again to remind myself of the exact technical details.

How is it done for other apps of this repo ? I mean all apps are currently deployed as runner jars ...

**Comment by manstis** [comment]
> > Without it, the JAR built by Quarkus, could not be used as a local container image... I'd have to dig again to remind myself of the exact technical details.
> 
> How is it done for other apps of this repo ? I mean all apps are currently deployed as runner jars ...

I suspect it is not done for any other apps. 

The only other scenario I know of is for [this](https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-deployment/src/main/java/org/kie/kogito/quarkus/processes/deployment/KogitoDevServicesBuildTimeConfig.java#L45-L46) but @cristianonicolai always uses that latest at quay.io.

IDK if he's been able to run that extension with a local build of [this](https://github.com/kiegroup/kogito-apps/tree/main/persistence-commons/quarkus-postgresql-inmemory-parent) that is packaged and published to quay.io [here](https://github.com/kiegroup/kogito-images/blob/main/modules/kogito-data-index-ephemeral/module.yaml#L5).

Could I ask that you comment out the line and add a comment above it stating why it has been removed (all be it temporarily) and ideally link to the JIRA you describe? I don't have a need to build and debug with it locally very often; but I have had the need. It'd be nice to have a simple reminder in `application.properties` than trawl through commit histories.

**Comment by radtriste** [comment]
sure will update that

**Comment by radtriste** [comment]
@manstis updated

File: trusty/trusty-service/trusty-service-postgresql/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Container image
 quarkus.container-image.build=${quarkus.build.image:true}
 quarkus.container-image.group=org.kie.kogito
-quarkus.package.type=fast-jar
+# We need the legacy-jar deployed to Nexus for kogito-images build
+# Follow-up in https://issues.redhat.com/browse/KOGITO-6525
+# quarkus.package.type=fast-jar
```
Comment (manstis) [inline trusty/trusty-service/trusty-service-postgresql/src/main/resources/application.properties:7]:
```suggestion
# If building and testing usage locally the packaging needs to be fast-jar 
# quarkus.package.type=fast-jar
```

**Review by manstis — APPROVED** [review]
Thank-you @radtriste.

I've made a small suggestion; but a comment is better than no comment.

:+1:

**Comment by radtriste** [comment]
FYI, https://issues.redhat.com/browse/KOGITO-6684 would allow to set back fast-jar on kogito-apps by default once implemented :)

**Comment by kie-ci1** [comment]
**(tests) - kogito-apps job** #588 was: **ABORTED**
Possible explanation: Most probably a timeout, please review


**Test results:**
- PASSED: 2402
- FAILED: 0

Those are the test failures: none

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/588/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-01-31T17:37:16.216Z] [INFO] Reading UCFGs from: /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-apps.tests.kogito-apps/bc/kiegroup_kogito_apps/kogito-apps/target/sonar/ucfg2/python<br/>[2022-01-31T17:37:16.216Z] [INFO] No UCFGs have been included for analysis.<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor PythonSecuritySensor [security] (done) | time=1ms<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor JsSecuritySensor [security]<br/>[2022-01-31T17:37:16.216Z] [INFO] Reading type hierarchy from: /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-apps.tests.kogito-apps/bc/kiegroup_kogito_apps/kogito-apps/target/sonar/ucfg2/js<br/>[2022-01-31T17:37:16.216Z] [INFO] Read 0 type definitions<br/>[2022-01-31T17:37:16.216Z] [INFO] Reading UCFGs from: /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-apps.tests.kogito-apps/bc/kiegroup_kogito_apps/kogito-apps/target/sonar/ucfg2/js<br/>[2022-01-31T17:37:16.216Z] [INFO] No UCFGs have been included for analysis.<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor JsSecuritySensor [security] (done) | time=0ms<br/>[2022-01-31T17:37:16.216Z] [INFO] ------------- Run sensors on project<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor Zero Coverage Sensor<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor Zero Coverage Sensor (done) | time=21ms<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor Java CPD Block Indexer<br/>[2022-01-31T17:37:16.216Z] [INFO] Sensor Java CPD Block Indexer (done) | time=509ms<br/>[2022-01-31T17:37:16.216Z] [INFO] CPD Executor 327 files had no CPD blocks<br/>[2022-01-31T17:37:16.216Z] [INFO] CPD Executor Calculating CPD for 926 files<br/>[2022-01-31T17:37:16.216Z] [INFO] CPD Executor CPD calculation finished (done) | time=310ms<br/>[2022-01-31T17:37:16.487Z] [INFO] SCM writing changed lines<br/>[2022-01-31T17:37:16.487Z] [INFO] SCM writing changed lines (done) | time=14ms<br/>[2022-01-31T17:37:16.791Z] [INFO] Analysis report generated in 331ms, dir size=2 MB<br/>[2022-01-31T17:37:17.349Z] Cancelling nested steps due to timeout<br/>[2022-01-31T17:37:17.352Z] Sending interrupt signal to process<br/>[2022-01-31T17:37:17.740Z] [INFO] Analysis report compressed in 942ms, zip size=1 MB<br/>[2022-01-31T17:37:18.302Z] + test 143 -eq 0<br/>[2022-01-31T17:37:18.307Z] script returned exit code 1<br/>[Pipeline] }<br/>[Pipeline] // withCredentials<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-01-31T17:37:19.036Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/588/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>[2022-01-31T17:37:20.435Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/588/consoleText<br/>[2022-01-31T17:37:20.435Z] + tail -n 50<br/>
</details>

**Pull request merged**